### PR TITLE
feature: ZENKO-732 Lifecycle transition policies

### PR DESCRIPTION
**Pull request by bennettbuchanan**
Add functional tests for lifecycle transition policies.

Depends on https://github.com/scality/Arsenal/pull/604.

### CODE REVIEW

**Comment by bennettbuchanan** [comment]
@bert-e wait

File: package.json [hunk 1]
```diff
@@ -19,9 +19,9 @@
   },
   "homepage": "https://github.com/scality/S3#readme",
   "dependencies": {
-    "arsenal": "github:scality/arsenal#3e08bad",
+    "arsenal": "github:scality/arsenal#0c78ccb",
     "async": "~2.5.0",
-    "aws-sdk": "2.28.0",
+    "aws-sdk": "2.60.0",
```
Comment (bennettbuchanan) [inline package.json]: Upgrading to this specific version to get the `Date` field bugfix. See: https://github.com/aws/aws-sdk-js/blob/HEAD/CHANGELOG.md#2600

**Review by tmacro — APPROVED** [review]

**Review by philipyoo — APPROVED** [review]
nice! lgtm

**Comment by bennettbuchanan** [comment]
Closing in favor of a branch name with correct Jira ticket. 
https://github.com/scality/cloudserver/pull/1653

**Pull request closed without merging**